src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -726,86 +753,112 @@ DatabaseShardImp::onChildrenStopped()
+        return;
+
+    std::uint32_t earliestIndex = std::numeric_limits<std::uint32_t>::max();
+    std::uint32_t latestIndex = 0;
+    {
+        auto loadLedger =
+            [this](
+                char const* const sortOrder) -> std::optional<std::uint32_t> {
+            std::shared_ptr<Ledger> ledger;
+            std::uint32_t ledgerSeq;
+            std::tie(ledger, ledgerSeq, std::ignore) = loadLedgerHelper(
+                "WHERE LedgerSeq >= " + std::to_string(earliestLedgerSeq()) +
+                    " order by LedgerSeq " + sortOrder + " limit 1",
+                app_,
+                false);
+            if (!ledger || ledgerSeq == 0)
+            {
+                JLOG(j_.error()) << "no suitable ledgers were found in"
+                                    " the SQLite database to import";
+                return std::nullopt;
+            }
+            return ledgerSeq;
+        };
+
+        // Find earliest ledger sequence stored
+        auto const earliestLedgerSeq{loadLedger("asc")};
+        if (!earliestLedgerSeq)
             return;
-        }
+        earliestIndex = seqToShardIndex(*earliestLedgerSeq);
 
-        std::uint32_t earliestIndex;
-        std::uint32_t latestIndex;
-        {
-            auto loadLedger = [&](bool ascendSort =
-                                      true) -> boost::optional<std::uint32_t> {
-                std::shared_ptr<Ledger> ledger;
-                std::uint32_t ledgerSeq;
-                std::tie(ledger, ledgerSeq, std::ignore) = loadLedgerHelper(
-                    "WHERE LedgerSeq >= " +
-                        std::to_string(earliestLedgerSeq()) +
-                        " order by LedgerSeq " + (ascendSort ? "asc" : "desc") +
-                        " limit 1",
-                    app_,
-                    false);
-                if (!ledger || ledgerSeq == 0)
-                {
-                    JLOG(j_.error()) << "no suitable ledgers were found in"
-                                        " the SQLite database to import";
-                    return boost::none;
-                }
-                return ledgerSeq;
-            };
+        // Consider only complete shards
+        if (earliestLedgerSeq != firstLedgerSeq(earliestIndex))
+            ++earliestIndex;
 
-            // Find earliest ledger sequence stored
-            auto ledgerSeq{loadLedger()};
-            if (!ledgerSeq)
-                return;
-            earliestIndex = seqToShardIndex(*ledgerSeq);
+        // Find last ledger sequence stored
+        auto const latestLedgerSeq = loadLedger("desc");
+        if (!latestLedgerSeq)
+            return;
+        latestIndex = seqToShardIndex(*latestLedgerSeq);
 
-            // Consider only complete shards
-            if (ledgerSeq != firstLedgerSeq(earliestIndex))
-                ++earliestIndex;
+        // Consider only complete shards
+        if (latestLedgerSeq != lastLedgerSeq(latestIndex))
+            --latestIndex;
```

Comment:
`earliestIndex` and `latestIndex` can be made `const` by wrapping their initialization in lambdas. It could look something like:
```
    // Find earliest ledger sequence stored
    auto const earliestLedgerSeq{loadLedger("asc")};
    if (!earliestLedgerSeq)
        return;
    std::uint32_t const earliestIndex = [&] {
        std::uint32_t earliestIndex = seqToShardIndex(*earliestLedgerSeq);
        // Consider only complete shards
        if (earliestLedgerSeq != firstLedgerSeq(earliestIndex))
            ++earliestIndex;
        return earliestIndex;
    }();

    // Find last ledger sequence stored
    auto const latestLedgerSeq = loadLedger("desc");
    if (!latestLedgerSeq)
        return;
    std::uint32_t const latestIndex = [&] {
        std::uint32_t latestIndex = seqToShardIndex(*latestLedgerSeq);

        // Consider only complete shards
        if (latestLedgerSeq != lastLedgerSeq(latestIndex))
            --latestIndex;
        return latestIndex;
    }();
```

Reply:
Done